pkg/process/metadata/parser/service.go

```diff
@@ -364,3 +368,43 @@ func advancedGuessJavaServiceName(se *ServiceExtractor, process *procutil.Proces
 	}
 	return "", false
 }
+
+func parseCommandContextNodeJs(se *ServiceExtractor, process *procutil.Process, args []string) string {
+	if !se.useImprovedAlgorithm {
+		return "node"
+	}
+	skipNext := false
+	for _, a := range args {
+		if skipNext {
+			skipNext = false
+			continue
+		}
+		if strings.HasPrefix(a, "-") {
+			if a == "-r" || a == "--require" {
+				// next arg can be a js file but not the entry point. skip it
+				skipNext = !strings.ContainsRune(a, '=') // in this case the value is already in this arg
+				continue
+			}
+		} else if strings.HasSuffix(strings.ToLower(a), ".js") {
+			absFile := abs(filepath.Clean(a), process.Cwd)
+			if _, err := os.Stat(absFile); err == nil {
+				value, ok := nodejsparser.FindNameFromNearestPackageJSON(absFile)
+				if ok {
+					return value
+				}
+				break
+			}
+		}
+	}
+	return "node"
+}
+
+// abs returns the path itself is already absolute or the absolute path by joining cwd with path
```

Comment:
I think there's a typo
```suggestion
// abs returns the path itself if already absolute or the absolute path by joining cwd with path
```